apis/core/v1alpha2/core_types.go

```diff
@@ -504,6 +546,11 @@ type DataInput struct {
 
 	// ToFieldPaths specifies the field paths of an object to fill passed value.
 	ToFieldPaths []string `json:"toFieldPaths,omitempty"`
+
+	Conditions []ConditionRequirement `json:"conditions,omitempty"`
```

Comment:
Why add `Conditions` to dataInput?

Reply:
This is similar with conditions in dataOutput. These conditions are used to represent the data in intermediate storage is ready now and can be passed to other objects.